File: aJava/src/Busqueda_Binaria.java
public class Busqueda_Binaria {

    static int busquedaBinaria(int[] lista, int objetivo) {
        int bajo = 0;
        int alto = lista.length - 1;

        while (bajo <= alto) {
            int medio = (bajo + alto) / 2;
            if (lista[medio] == objetivo) {
                return medio;
            } else if (lista[medio] < objetivo) {
                bajo = medio + 1;
            } else {
                alto = medio - 1;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] numeros = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        int posicion = busquedaBinaria(numeros, 7);
        System.out.println("El elemento 7 esta en la posicion " + posicion);
    }
}

File: Python/Busqueda_Binaria.py
def busqueda_binaria(lista, objetivo):
    bajo = 0
    alto = len(lista) - 1

    while bajo <= alto:
        medio = (bajo + alto) // 2
        if lista[medio] == objetivo:
            return medio
        elif lista[medio] < objetivo:
            bajo = medio + 1
        else:
            alto = medio - 1

    return -1

numeros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
posicion = busqueda_binaria(numeros, 7)
print(f"El elemento 7 esta en la posicion {posicion}")